### PR TITLE
Implements IOB tile merge

### PR DESCRIPTION
**Pull request by litghost**
This implements the IOB into IOI tile merge, and adds an initial tile for the merged tile.  This tile is missing internal connections required to route.

### CODE REVIEW

File: xc7/archs/artix7/devices/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
   ARCH artix7
+  USE_ROI
   DEVICES xc7a50t-basys3 xc7a50t-arty-uart xc7a50t-arty-swbut
   PARTS xc7a35tcpg236-1 xc7a35tcsg324-1 xc7a35tcsg324-1
 )
+add_xc7_device_define(
+  ARCH artix7
+  DEVICES xc7a50t
+  PARTS xc7a35tcpg236-1
+  GRAPH_LIMIT 61,105,114,155
```
Comment (mithro) [inline xc7/archs/artix7/devices/CMakeLists.txt]: I don't like this format very much -- but I assume it is only temporary? If so I don't care that much...

File: xc7/utils/prjxray_tile_import.py [hunk 2]
```diff
@@ -696,6 +697,278 @@ def import_site_as_tile(db, args):
...
+    site_type_count = {}
+    site_prefixes = {}
+    cells_idx = []
+
+    ignored_site_types = set()
+
+    site_type_ports = {}
+    cur.execute(
+        """
+SELECT DISTINCT site_type.name, site.x_coord, site.y_coord
```
Comment (mithro) [inline xc7/utils/prjxray_tile_import.py]: Style of this SQL statement doesn't match the others?

File: xc7/utils/prjxray_tile_import.py [hunk 3]
```diff
@@ -696,6 +697,278 @@ def import_site_as_tile(db, args):
...
+                metadata_xml, XI_INCLUDE, {
+                    'href': site_type_path,
+                    'xpointer': "xpointer(pb_type/metadata/child::node())",
+                }
+            )
+
+    # Iterate over sites in tile
+    cur.execute(
+        """
+SELECT site_type.name, site.pkey
```
Comment (mithro) [inline xc7/utils/prjxray_tile_import.py]: Same?

File: xc7/utils/prjxray_tile_import.py [hunk 4]
```diff
@@ -696,6 +697,278 @@ def import_site_as_tile(db, args):
...
+        site_idx = cells_idx[idx]
+        site_instance = site_type_instances[site_type][site_idx]
+        site_name = cell_names[site_instance]
+
+        interconnect_xml.append(ET.Comment(" Tile->Site "))
+
+        # Iterate over pins in site
+        cur2.execute(
+            """
+SELECT site_pin.name, site_pin.direction, wire_in_tile.name
```
Comment (mithro) [inline xc7/utils/prjxray_tile_import.py]: Same?

File: xc7/utils/prjxray_tile_import.py [hunk 5]
```diff
@@ -757,6 +1030,11 @@ def main():
         "Typically a tile can treat the sites within the tile as independent.  For tiles where this is not true, fused sites only imports 1 primatative for the entire tile, which should be named the same as the tile type."
     )
 
+    parser.add_argument(
+        '--connection_database',
+        help="Location of connection database to use in lue of Project X-Ray",
```
Comment (mithro) [inline xc7/utils/prjxray_tile_import.py]: I don't understand what this means?

File: xc7/archs/artix7/devices/CMakeLists.txt (hunk 1 above)
Comment (litghost) [reply xc7/archs/artix7/devices/CMakeLists.txt]: Ya, it's just temporary.  This will go away after we support the full 50T routing graph.

**Pull request merged**